### PR TITLE
Fix UAFs and NULL DEREFs reported by Coverity

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Just scoured through high-impact issues reported by Coverity.
Can be reviewed commit by commit.

*Note, that I marked UAF fixes with (UAF), OOB with (OOB) and NULL DEREFS in the commit message and they can be cherry-picked for 0.2.x patch release.*

**Test plan**

CI is green

### CODE REVIEW

File: librz/asm/asm.c [hunk 1]
```diff
@@ -321,7 +321,7 @@ RZ_API bool rz_asm_use(RzAsm *a, const char *name) {
 		return true;
 	}
 	rz_list_foreach (a->plugins, iter, h) {
-		if (!strcmp(h->name, name) && h->arch) {
+		if (h && h->arch && h->name && !strcmp(h->name, name)) {
```
Comment (ret2libc) [inline librz/asm/asm.c]:
```suggestion
		if (h->arch && h->name && !strcmp(h->name, name)) {
```
I think `h` should not be checked. If `h` is NULL there's definitely something "wronger" somewhere else.

File: librz/reg/profile.c [hunk 2]
```diff
@@ -397,7 +397,6 @@ RZ_API char *rz_reg_parse_gdb_profile(const char *profile_file) {
 				str = rz_file_slurp(file, NULL);
 				free(file);
 			}
-			free(base);
```
Comment (ret2libc) [inline librz/reg/profile.c]: ?? why?

File: librz/asm/asm.c (hunk 1 above)
Comment (XVilka) [reply librz/asm/asm.c]: We can add `rz_return_if_fail()` then here.

File: librz/reg/profile.c (hunk 2 above)
Comment (XVilka) [reply librz/reg/profile.c]: image

Comment (XVilka) [reply librz/reg/profile.c]: Though you are probably right and it's false positive in this concrete case, I will remove this commit.

File: librz/asm/asm.c (hunk 1 above)
Comment (ret2libc) [reply librz/asm/asm.c]: I would just remove that completely. If you have doubts, have a look at the rz_list_append that append plugins to `a->plugins`.

Comment (XVilka) [reply librz/asm/asm.c]: Removed NULL check for `h` here.

**Review by wargio — APPROVED** [review]

**Pull request merged**